File: SimpleActivity/app/src/main/java/ao/uan/fc/dam/simpleactivity/MainActivity.java
package ao.uan.fc.dam.simpleactivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;


public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

   @Override
   public void onCreate(Bundle savedInstanceState) {
       super.onCreate(savedInstanceState);
       EdgeToEdge.enable(this);
       setContentView(R.layout.activity_main);

       Log.i(TAG, "onCreate");
   }

   @Override
   public void onStart() {
       super.onStart();
       Log.i(TAG, "onStart");
   }

   @Override
   public void onResume() {
       super.onResume();
       Log.i(TAG, "onResume");
       Toast.makeText(this, "onResume", Toast.LENGTH_SHORT).show();
   }

   @Override
   public void onPause() {
       super.onPause();
       Log.i(TAG, "onPause");
   }

   @Override
   public void onStop() {
       super.onStop();
       Log.i(TAG, "onStop");
   }

   @Override
   public void onDestroy() {
       super.onDestroy();
       Log.i(TAG, "onDestroy");
   }

   @Override
   public void onRestart() {
       super.onRestart();
       Log.i(TAG, "onRestart");
   }

    public void finishButtonPressed(View view) {
        finish();
    }
}

File: HelloWorld/app/src/main/java/ao/uan/fc/dam/exercise_1/MainActivity.kt
package ao.uan.fc.dam.exercise_1

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity

// Constante para a chave da mensagem (equivalente ao static final em Java)
const val EXTRA_MESSAGE = "ao.uan.fc.dam.MESSAGE"

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btnSend = findViewById<Button>(R.id.BtnSend)
        val editText = findViewById<EditText>(R.id.editTextMessage)

        btnSend.setOnClickListener {
            val message = editText.text.toString()

            // Criar a Intent para a DisplayMessageActivity
            val intent = Intent(this, DisplayMessageActivity::class.java).apply {
                putExtra(EXTRA_MESSAGE, message)
            }
            startActivity(intent)
        }
    }

    /** Chamado quando o usuário clica no botão Send */
    fun sendMessage(view: View) {
        val editText = findViewById<EditText>(R.id.editTextMessage)
        val message = editText.text.toString()

        // Intent para iniciar a segunda Activity
        val intent = Intent(this, DisplayMessageActivity::class.java).apply {
            putExtra(EXTRA_MESSAGE, message)
        }
        startActivity(intent)
    }
}